signalfx-transform-helper.js

```diff
@@ -1,36 +1,30 @@
 'use strict';
 
-const FIELD_SEPARATOR = '_';
 const MAX_FIELDNAME_LENGTH = 128;
+const DETAIL_PREFIX = "detail_";
 
 function isPrimitive(x) {
   return Object(x) !== x;
 }
 
 function sanitize(name) {
-  let sanitizedName = name.replace(/[^a-zA-Z0-9\-_]+/gi, '_');
+  let sanitizedName = name.replace(/[^a-zA-Z0-9\-_]+/gi, '');
```

Comment:
as I recall, sometimes we get the field names with spaces. Wouldn't it be better to stick to the previous convention with underscore instead of concatenating them without any separator? To me this change unnecessarily breaks the behavior even for simple fields and objects.